### PR TITLE
react/Week3 homework added

### PR DESCRIPTION
**Pull request by seenachacko**

### CODE REVIEW

File: react/week3/todo-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    console.log(startDate);
+    setTodoList([
+      ...todoList,
+      {
+        id: todoList.length > 0 ? todoList[todoList.length - 1].id + 1 : 1,
+        description: userInput,
+        deadline: startDate.toISOString().substring(0, 10),
+      },
+    ]);
+    console.log(todoList);
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.js]: remove all console.log

File: react/week3/todo-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import Header from "./Components/Header";
+import "./App.css";
+import Addtodo from "./Components/Addtodo";
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.js]:
```suggestion
import AddTodo from "./Components/AddTodo";
```

File: react/week3/todo-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        item.description = newdescription;
+      }
+      return item;
+    });
+  };
+  return (
+    <div className="main-Container">
+      <Header />
+      <Timer />
+      <Addtodo addTodo={addTodo} />
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.js]:
```suggestion
      <AddTodo addTodo={addTodo} />
```

File: react/week3/todo-app/src/Components/Addtodo.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Addtodo.js:1]: the file name should be AddTodo.js

File: react/week3/todo-app/src/Components/Border.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+const Border = ({ children }) => {
+  return (
+    <div style={{ border: "2px solid white", margin: "5px" }}>{children}</div>
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Border.js:4]: I would add a class name and define style for it in CSS file

File: react/week3/todo-app/src/App.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+.add-button,
+.delete-button,
+.edit-button,
+.update-button {
+  margin: 5px;
+  background: green;
+  color: white;
+  border-radius: 5px;
+}
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.css]:
```suggestion
button {
  margin: 5px;
  background: green;
  color: white;
  border-radius: 5px;
}
```

File: react/week3/todo-app/src/App.css [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  margin: 5px;
+}
+ul,
+li {
+  list-style: none;
+  color: indigo;
+  font-weight: bold;
+  text-align: center;
+}
+.header-Container {
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.css]:
```suggestion
.header-container {
```
kebab-case (dash-case) convention for classes names

File: react/week3/todo-app/src/App.css [hunk 8]
```diff
@@ -0,0 +1,40 @@
+.App {
+  text-align: center;
+}
+.main-Container {
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.css]:
```suggestion
.main-container {
```
kebab-case

File: react/week3/todo-app/src/App.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+import Header from "./Components/Header";
+import "./App.css";
+import Addtodo from "./Components/Addtodo";
+import Timer from "./Components/Timer";
+import Todolist from "./Components/TodoList";
+function App() {
+  const [error, setError] = useState(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [todoList, setTodoList] = useState([]);
+  const Api =
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.js]:
```suggestion
  const api =
```

File: react/week3/todo-app/src/App.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+  useEffect(() => {
+    fetch(Api)
```
Comment (cecastosic) [inline react/week3/todo-app/src/App.js]:
```suggestion
    fetch(api)
```

File: react/week3/todo-app/src/Components/Todo.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        </button>
+        <button className="delete-button" onClick={() => deleteTodo(toDo.id)}>
+          delete
+        </button>
+      </li>
+    </>
+  );
+
+  const viewTemplate = (
+    <>
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
    <li>
```
only li can be a child element of ul, please check HTML structure

File: react/week3/todo-app/src/Components/TodoList.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import Todo from "./Todo";
+function Todolist({ toDoList, deleteTodo, updateTodo }) {
+  return (
+    <div>
+      <ul>
+        {toDoList.length > 0
+          ? toDoList.map((item) => (
+              <Todo
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/TodoList.js]: Todo has to start with li tag

File: react/week3/todo-app/src/Components/Todo.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+      <Border>
+        <li
+          className="todo"
+          style={{ textDecoration: status ? "line-through" : "none" }}
+        >
+          {toDo.description} - {toDo.deadline}
+          <input type="checkbox" value={toDo.id} onClick={check}></input>
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
          <input type="checkbox" value={todo.id} onClick={check}></input>
```

File: react/week3/todo-app/src/Components/Todo.js [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+  );
+
+  const viewTemplate = (
+    <>
+      <Border>
+        <li
+          className="todo"
+          style={{ textDecoration: status ? "line-through" : "none" }}
+        >
+          {toDo.description} - {toDo.deadline}
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
          {todo.description} - {todo.deadline}
```

File: react/week3/todo-app/src/Components/Todo.js [hunk 15]
```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+import Border from "./Border";
+import PropTypes from "prop-types";
+function Todo({ toDo, deleteTodo, updateTodo }) {
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
function Todo({ todo, deleteTodo, updateTodo }) {
```

File: react/week3/todo-app/src/Components/TodoList.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+                key={item.id}
+                toDo={item}
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/TodoList.js]:
```suggestion
                todo={item}
```

File: react/week3/todo-app/src/Components/TodoList.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import Todo from "./Todo";
+function Todolist({ toDoList, deleteTodo, updateTodo }) {
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/TodoList.js]:
```suggestion
function Todolist({ todoList, deleteTodo, updateTodo }) {
```

File: react/week3/todo-app/src/Components/Todo.js [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+  const viewTemplate = (
+    <>
+      <Border>
+        <li
+          className="todo"
+          style={{ textDecoration: status ? "line-through" : "none" }}
+        >
+          {toDo.description} - {toDo.deadline}
+          <input type="checkbox" value={toDo.id} onClick={check}></input>
+          <button className="delete-button" onClick={() => deleteTodo(toDo.id)}>
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
          <button className="delete-button" onClick={() => deleteTodo(todo.id)}>
```

File: react/week3/todo-app/src/Components/Todo.js [hunk 19]
```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+import Border from "./Border";
+import PropTypes from "prop-types";
+function Todo({ toDo, deleteTodo, updateTodo }) {
+  const [status, setStatus] = useState(false);
+  const [isEditing, setEditState] = useState(false);
+  const [newdescription, setDescription] = useState(toDo.description);
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
  const [newDescription, setNewDescription] = useState(toDo.description);
```
https://reactjs.org/docs/hooks-state.html

File: react/week3/todo-app/src/Components/Todo.js [hunk 20]
```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+import Border from "./Border";
+import PropTypes from "prop-types";
+function Todo({ toDo, deleteTodo, updateTodo }) {
+  const [status, setStatus] = useState(false);
+  const [isEditing, setEditState] = useState(false);
```
Comment (cecastosic) [inline react/week3/todo-app/src/Components/Todo.js]:
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```
https://reactjs.org/docs/hooks-state.html

**Review by cecastosic — COMMENTED** [review]
Good job in general! 👍  Be careful with naming in general and following naming conventions. Good reading https://www.robinwieruch.de/javascript-naming-conventions
Always remove console.log before making PR 🙂 
And be aware of HTML structure and valid markup.

File: react/week3/todo-app/src/Components/Todo.js (hunk 20 above)
Comment (seenachacko) [reply react/week3/todo-app/src/Components/Todo.js]: Thanks for your feedbacks, Time and good patience.:heart_eyes:
sorry for making a lot of naming error. 
I will take care from next time 👍

**Pull request merged**